File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/secret/SecretModule.java
package at.tugraz.tc.cyfile.secret;

import javax.inject.Singleton;

import at.tugraz.tc.cyfile.crypto.KeyVaultService;
import dagger.Module;
import dagger.Provides;

@Module
public class SecretModule {

    private final SecretManager secretManager;

    private final SecretPrompter secretPrompter;

    private final KeyVaultService keyVaultService;

    private final AppHidingModule appHidingModule;

    public SecretModule(SecretManager secretManager, SecretPrompter secretPrompter,
                        KeyVaultService keyVaultService, AppHidingModule appHidingModule) {
        this.secretManager = secretManager;
        this.secretPrompter = secretPrompter;
        this.keyVaultService = keyVaultService;
        this.appHidingModule = appHidingModule;
    }

    @Provides
    @Singleton
    public SecretManager providesSecretManager() {
        return secretManager;
    }

    @Provides
    @Singleton
    public SecretPrompter providesSecretPrompter() {
        return secretPrompter;
    }

    @Provides
    @Singleton
    public KeyVaultService providesKeyVaultService() {
        return keyVaultService;
    }

    @Provides
    @Singleton
    public AppHidingModule providesAppHidingModule() {
        return appHidingModule;
    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/secret/SecretModule.kt
package at.tugraz.tc.cyfile.secret

import javax.inject.Singleton

import at.tugraz.tc.cyfile.crypto.KeyVaultService
import dagger.Module
import dagger.Provides

@Module
class SecretModule(private val secretManager: SecretManager, private val secretPrompter: SecretPrompter, private val keyVaultService: KeyVaultService) {

    @Provides
    @Singleton
    fun providesSecretManager(): SecretManager {
        return secretManager
    }

    @Provides
    @Singleton
    fun providesSecretPrompter(): SecretPrompter {
        return secretPrompter
    }

    @Provides
    @Singleton
    fun providesKeyVaultService(): KeyVaultService {
        return keyVaultService
    }
}
